src/components/form/submission.vue

```diff
@@ -73,7 +78,14 @@ const fetchForm = () => {
       (problem.code === 404.1 ? t('formNotFound') : null)
   }).then(() => {
     if (form.data.webformsEnabled) {
-      // TODO: maybe change the route if it is /f/... and it is for "New / Edit Submission"
+      // change the route if it is /f/... and it is for "New / Edit Submission"
+      if (route.path.startsWith('/f/')) {
+        if (!actionType && !form.data.draftToken) {
+          router.replace(formPath(form.data.projectId, form.data.xmlFormId, 'submissions/new'));
+        } else if (actionType === 'edit' && route.query.instance_id) {
```

Comment:
This is probably a question for later, but how would Web Forms work for editing if there's no `instance_id` query parameter? Instead of trying to render `WebFormRenderer`, maybe we should show an alert or render `NotFound` in that case.